src/main/java/io/cdap/plugin/zendesk/connector/ZendeskConnectorConfig.java

```diff
@@ -0,0 +1,124 @@
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * ZendeskConnectorConfig Class
+ */
+public class ZendeskConnectorConfig extends PluginConfig {
+
+  public static final String PROPERTY_ADMIN_EMAIL = "adminEmail";
+  public static final String PROPERTY_API_TOKEN = "apiToken";
+  public static final String PROPERTY_SUBDOMAINS = "subdomains";
+  private static final String PATH_SEGMENT = "https://%s.zendesk.com/api/v2/%s";
+  private static final Logger LOG = LoggerFactory.getLogger(ZendeskConnectorConfig.class);
+
+  @Name(PROPERTY_ADMIN_EMAIL)
+  @Description("Zendesk admin email.")
+  @Macro
+  private final String adminEmail;
+  @Name(PROPERTY_API_TOKEN)
+  @Description("Zendesk API token.")
+  @Macro
+  private final String apiToken;
+  @Name(PROPERTY_SUBDOMAINS)
+  @Description("Zendesk Subdomains to read objects from.")
+  @Macro
+  private final String subdomains;
+
+  public ZendeskConnectorConfig(String adminEmail, String apiToken, String subdomains) {
+    this.adminEmail = adminEmail;
+    this.apiToken = apiToken;
+    this.subdomains = subdomains;
+  }
+
+  public String getAdminEmail() {
+    return adminEmail;
+  }
+
+  public String getApiToken() {
+    return apiToken;
+  }
+
+  public Set<String> getSubdomains() {
+    return getList(subdomains);
+  }
+
+  protected Set<String> getList(String value) {
+    return Strings.isNullOrEmpty(value)
+      ? Collections.emptySet()
+      : Stream.of(value.split(","))
+      .map(String::trim)
+      .filter(name -> !name.isEmpty())
+      .collect(Collectors.toSet());
+  }
+
+  public void validateConnectionParameters(FailureCollector collector) throws IOException {
+    if (containsMacro(PROPERTY_ADMIN_EMAIL)
+      || containsMacro(PROPERTY_API_TOKEN)
+      || containsMacro(PROPERTY_SUBDOMAINS)
+      || containsMacro(ZendeskBatchSourceConfig.PROPERTY_MAX_RETRY_COUNT)
+      || containsMacro(ZendeskBatchSourceConfig.PROPERTY_CONNECT_TIMEOUT)
+      || containsMacro(ZendeskBatchSourceConfig.PROPERTY_READ_TIMEOUT)) {
+      return;
+    }
+    getSubdomains().forEach(subdomain -> {
+      String countURL = String.format(PATH_SEGMENT, subdomain, ObjectType.GROUPS.getApiEndpoint());
+      HttpClientContext httpClientContext = HttpUtil.createHttpContext(this, countURL);
+      HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+      CloseableHttpClient httpClient = httpClientBuilder.build();
+      CloseableHttpResponse response = null;
+      try {
+        response = httpClient.execute(new HttpGet(countURL), httpClientContext);
+      } catch (IOException e) {
+        LOG.error("Unable to generate response", e);
+      }
+      StatusLine statusLine = response.getStatusLine();
```

Comment:
if code will throw IOException, there will null response and will get NullPointerException here. Add error message in FailureCollector inside catch block and return from there.

Reply:
Done